### ISSUE TITLE
Waste Section not full width and dropping

### ISSUE THREAD

**Issue by walpoletim-bjss** [textiles-lab/knitout-backend-kniterate#34]
Hi

We are using knitsketching  to generate knitout code, and when we  add waste yarn to this, it does not add the full width, and when you run this, it drops the waste yarn as it starts the knit.

.k file attached as an example

Any ideas?

[beanie.k.txt](https://github.com/textiles-lab/knitout-backend-kniterate/files/7055372/beanie.k.txt)
[beanie-kniterate-waste.k.txt](https://github.com/textiles-lab/knitout-backend-kniterate/files/7055548/beanie-kniterate-waste.k.txt)

<img width="736" alt="Screenshot 2021-08-26 at 09 48 39" src="https://user-images.githubusercontent.com/10545041/130932325-14d5dd67-cad9-498e-a394-3db8d46a768e.png">

**Comment by ixchow** [comment]
Yep, looks like beanie-kniterate-waste.k.txt is dropping all of the loops added (lines 1707 - 1745).

How did you go about adding the waste yarn? If it was created with `waste-section.js` -- which is definitely more of a helpful hack than a supported part of the backend -- it would be helpful to know what options you passed to the program. (Trying it myself, I can't get it to do anything other than throw an exception!)

**Comment by walpoletim-bjss** [comment]
Hi - I send the original .k file which also needs converting to kniterate format with the kniterate knitout-alter-kniterate.js.  I have also had to remove some invalid x-stitch-numbers as they were incorrect.

New files uploaded
Knitout - Kniterate
[beanie-kniterate.k.txt](https://github.com/textiles-lab/knitout-backend-kniterate/files/7072373/beanie-kniterate.k.txt)
Knitout - Pre conversation to Knitout
[beanie1.k.txt](https://github.com/textiles-lab/knitout-backend-kniterate/files/7072374/beanie1.k.txt)
Knitout - Kniterate - Waste 
[beanie-kniterate-waste.k.txt](https://github.com/textiles-lab/knitout-backend-kniterate/files/7072376/beanie-kniterate-waste.k.txt)

Waste paramaters
<img width="984" alt="Screenshot 2021-08-29 at 12 35 34" src="https://user-images.githubusercontent.com/10545041/131249022-0faf4534-7def-444f-818d-08e11ee05512.png">

**Comment by ixchow** [comment]
I can confirm that the parameters above do duplicate the bug. I suspect the problem is with the width determination code in `waste-section.js`: https://github.com/textiles-lab/knitout-backend-kniterate/blob/b19ae3d25fe7b977513362731ed1ff9220dacf69/extras/waste-section.js#L210-L261

This code looks suspiciously complicated and also looks like it probably bails out (via `break findMinMax`) earlier than it should. I suspect the resolution to this issue lies in re-writing this section of code. (Or, perhaps, re-writing the utility altogether, since some additional options might make sense for -- e.g. -- half-gauge patterns.)

Unfortunately, we don't at present have anyone actively working on the `extras/` folder (or, really, the backend in general), so I would anticipate that this bug will remain open until someone sufficiently interested and helpful shows up.
